src/Logary.Prometheus/Exporter.fs

```diff
@@ -0,0 +1,171 @@
+﻿namespace Logary.Prometheus
+open System
+open System.Text
+open System.Text.RegularExpressions
+open Logary.Metric
+open Suave
+open Suave.Filters
+open Suave.Operators
+open Suave.Writers
+open Hopac
+
+
+type ExporterConf =
+  {
+    metricNameTransformer: string -> string
+    metricRegistry: MetricRegistry
+    webConfig: SuaveConfig
+    urlPath: string
+  }
+
+module ExporterConf =
+  let validNameRegex = Regex("^[a-zA-Z_][a-zA-Z0-9_]*$", RegexOptions.Compiled)
+  let reservedNameRegex =  Regex("^__.*$", RegexOptions.Compiled)
+  let startCharValidRegex =  Regex("^[a-zA-Z_].*$", RegexOptions.Compiled)
+  let notValidAfterPartRegex =  Regex("[^a-zA-Z0-9_]", RegexOptions.Compiled)
+
+  let defaultMetricNameTrans metricName =
+    let mutable nameResult = metricName
+    if String.IsNullOrEmpty nameResult then failwith "name can't be null"
+    if reservedNameRegex.IsMatch nameResult then
+      nameResult <- "metric_" + nameResult.Substring 2
+    if not <| startCharValidRegex.IsMatch nameResult then
+      nameResult <- "metric_" + nameResult
+    notValidAfterPartRegex.Replace(nameResult, "_")
+
+  let nameValidator name =
```

Comment:
Could this validator be named `validateName` return `Result.Ok`/`Result.Error` and then have another, `forceValidateName` that does `Result.getOrThrow`? Just to keep the default case total and the "easy" case throwing; aiming to have composable APIs.

Reply:
👍